docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/nci-promo-block/cgdp-promo-block.ts

```diff
@@ -0,0 +1,70 @@
+import { landingClickTracker } from '../../core/analytics/landing-page-contents-helper';
+
+/**
+ * Gets the title of the PromoBlock
+ * @param {HTMLElement} anchorEl - Clicked anchor element.
+ */
+const getTitle = (element: HTMLElement): string =>
+	element.querySelector('.nci-promo-block__heading')?.textContent || '_ERROR_';
+
+/**
+ * Gets the theme of the PromoBlock (Light / Dark)
+ * @param {HTMLElement} element - The Promo Block
+ */
+const getPromoTheme = (element: HTMLElement): string =>
+	element.classList.contains('nci-promo-block--dark') ? 'Dark' : 'Light';
+
+/**
+ * Gets the variant of the PromoBlock (No Image / Image Left / Image Right)
+ * @param {HTMLElement} element - The Promo Block
+ */
+const getPromoVariant = (element: HTMLElement): string => {
+	if (element.classList.contains('nci-alternating-right')) {
+		return 'Image Right';
+	} else if (element.classList.contains('nci-promo-block--with-image')) {
+		return 'Image Left';
+	} else return 'No Image';
+};
+
+/**
+ * Click handler for the language toggle click.
+ * @param promoBlockRow the promo-block element.
+ */
+const promoBlockLinkClickHandler =
+	(promoBlock: HTMLElement) => (evt: Event) => {
+		const target = evt.currentTarget as HTMLElement;
+
+		landingClickTracker(
+			target,
+			'PromoBlock', // linkName
+			1, // rowItems
+			1, // rowItemsIndex
+			'Promo Block', // componentType
+			getPromoTheme(promoBlock) as string, // componentTheme
+			getPromoVariant(promoBlock) as string, // componentVariant
+			getTitle(promoBlock) as string, // title
+			target.dataset.eddlLandingItemLinkType || '_ERROR_', // linkType
+			target.textContent?.trim() || '_ERROR_', // linkText
```

Comment:
Missing coverage can be fixed with

```suggestion
			!target.textContent ? '_ERROR_' : target.textContent.trim(), // linkText
```